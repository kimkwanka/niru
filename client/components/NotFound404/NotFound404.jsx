/* eslint-disable no-param-reassign */
import React from 'react';
import Helmet from 'react-helmet';
import PropTypes from 'prop-types';

const NotFound404 = ({ staticContext }) => {
  // Set StaticRouter context status to 404 to let the server know
  staticContext.status = 404;

  return (
    <div className="container margin-top-small">
      <Helmet title="Not Found" />
      <h1>404</h1>
      <p>Oops! Nothing here :(</p>
    </div>
  );
};

NotFound404.propTypes = {
  staticContext: PropTypes.objectOf(PropTypes.shape),
};

NotFound404.defaultProps = {
  staticContext: {},
};

export default NotFound404;
