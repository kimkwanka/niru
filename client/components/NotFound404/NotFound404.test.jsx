import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NotFound404 from './NotFound404';

describe('<NotFound404 /> - Snapshot', () => {
  it('matches its snapshot', () => {
    const wrapper = shallow(<NotFound404 />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
