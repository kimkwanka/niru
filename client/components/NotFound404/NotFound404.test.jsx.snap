// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound404 /> - Snapshot matches its snapshot 1`] = `
<div
  className="container margin-top-small"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Not Found
    </title>
  </HelmetWrapper>
  <h1>
    404
  </h1>
  <p>
    Oops! Nothing here :(
  </p>
</div>
`;
